### ISSUE TITLE
Adding Squarespace's dictionary to the compiler

### ISSUE THREAD

**Issue by jasonbarone** [Squarespace/template-compiler#2]
Hey, I've been playing with the compiler and it seems to work OK if I give it the Squarespace template file and its corresponding JSON, but it's returning errors on Squarespace-specific tags like `{.main-image?}`, `{.equal?`, etc. Have you guys open sourced your dictionary as well?

**Comment by phensley** [comment]
@jasonbarone The `main-image?` predicate lives in the plugins/squarespace project, in ContentPredicates.java, and `equal?` lives in the core, in CorePredicates.java.

These weren't added to the command line compiler by default, but I just corrected that -- pull `master` branch and run the following command:

    gradlew makeCli
    ./templatec -h

**Comment by jasonbarone** [comment]
@phensley Thanks man that definitely got it moving a little more. The next thing I'm seeing is that Squarespace block partials (in /blocks) are "not found". ie: let's say I'm running the compiler on `foo.region` and `foo.json`, and the .region is applying partials located in `/blocks`, do the compiler have to be configured for that? Basically I'm trying to compile a standard Squarespace template, and referencing cached JSON files. Java isn't my `forte` yet :-)

**Comment by phensley** [comment]
The use partials, the compiler needs access to a JSON object that contains a mapping from `name -> partial`. Any time it encounters an `apply <name>` it looks up that name in the partials set, compiles the partial and executes it.

For example, to execute `{@|apply foo}` you would need to provide the compiler a JSON tree like:  `{"foo": "<b>this is the partial to be rendered</b>"}`

You can see this using the command line compiler `templatec` and the `place` example:

    ./gradlew makeCli
    cd ./example
    ./templatec --partials place.partial --json place.json place.html

To do anything more advanced you may have to write a bit of Java.  The `TemplateC.java` class is what drives the `templatec` command, so that would provide a good starting point.

**Comment by jasonbarone** [comment]
Oh ok this makes a lot of sense. I had assumed that it mirrored the behavior of how Squarespace compiles a template. I did play with the example and didn't understand it until your comment here. I'll play around a bit more. I just tried out a few ideas of grabbing all of my block partials, turning the contents into a json string and adding it to the reference .json file. That actually worked fine. For example:

**place.html**
```
{.section place}
    In {name} the temperature is {@|apply show-temp} {@|apply another-example}
{.end}
```

**place.json**
```
{
    "place": {
        "name": "Timbuktu",
        "temperature": 123.7
    },
    "show-temp": "{temperature} degrees fahrenheit",
    "another-example": "example {temperature} here"
}
```

Then running:
```
templatec --partials place.json --json place.json place.html
```

Not sure if there's any reason why I shouldn't do that, but it gives me something without diving into Java :)
